### PR TITLE
fix: honor OPENCODE_DISABLE_CLAUDE_CODE env var

### LINKED ISSUE #2037: Claude plugin commands leak into OpenCode despite OPENCODE_DISABLE_CLAUDE_CODE*
## Bug Description

oh-my-opencode bypasses OpenCode's `OPENCODE_DISABLE_CLAUDE_CODE*` env vars by independently reading `~/.claude/plugins/installed_plugins.json` and injecting Claude Code components into the resolved config — without checking `process.env`.

OpenCode core already respects these env vars and disables Claude Code integration when they are set (confirmed in [anomalyco/opencode#10383](https://github.com/anomalyco/opencode/issues/10383)). The issue is exclusively in oh-my-opencode's independent Claude plugin loader path.

## Environment

- **OpenCode**: `1.2.10`
- **oh-my-opencode**: `3.8.1`
- **OS**: macOS (Darwin)

## Steps to Reproduce

1. Have any Claude Code plugins installed (e.g., `everything-claude-code`, `frontend-design@claude-plugins-official`).
2. Include `"oh-my-opencode@latest"` in OpenCode's `plugin` array.
3. Set env vars:
   ```bash
   export OPENCODE_DISABLE_CLAUDE_CODE=1
   export OPENCODE_DISABLE_CLAUDE_CODE_PROMPT=1
   export OPENCODE_DISABLE_CLAUDE_CODE_SKILLS=1
   export OPENCODE_DISABLE_EXTERNAL_SKILLS=1
   ```
4. Run:
   ```bash
   opencode debug config | rg "everything-claude-code|frontend-design"
   ```

## Expected Behavior

Claude Code plugin components should not appear in OpenCode's resolved config when `OPENCODE_DISABLE_CLAUDE_CODE=1` is set.

## Actual Behavior

Claude Code plugin entries (commands, skills, agents) still appear in the resolved config.

**Disabling oh-my-opencode** or setting `claude_code.*: false` in `oh-my-opencode.json` stops the leak — confirming the source.

### Zero integration with `OPENCODE_DISABLE_*` env vars

Searching the oh-my-opencode codebase for `OPENCODE_DISABLE` returns **zero results**. The plugin never reads `process.env` for these variables.

### 7 opt-out gates default to `true`

All Claude Code component loading uses `pluginConfig.claude_code?.X ?? true`. When `claude_code` is absent from config (the default), all gates resolve to `true`:

| File | Gate | Component |
|------|------|-----------|
| `src/plugin-handlers/plugin-components-loader.ts` | `claude_code?.plugins ?? true` | Plugin discovery from `~/.claude/plugins/` |
| `src/plugin-handlers/command-config-handler.ts` | `claude_code?.commands ?? true` | Commands |
| `src/plugin-handlers/command-config-handler.ts` | `claude_code?.skills ?? true` | Skills (command context) |
| `src/plugin-handlers/agent-config-handler.ts` | `claude_code?.skills ?? true` | Skills (agent awareness) |
| `src/plugin-handlers/agent-config-handler.ts` | `claude_code?.agents ?? true` | Agents from `~/.claude/agents/` |
| `src/plugin-handlers/mcp-config-handler.ts` | `claude_code?.mcp ?? true` | MCP servers |
| `src/plugin/hooks/create-transform-hooks.ts` | `claude_code?.hooks ?? true` | Hooks |

`src/plugin/skill-context.ts` uses an equivalent variant: `claude_code?.skills !== false`.

## Workaround

```json
{
  "claude_code": {
    "agents": false,
    "commands": false,
    "hooks": false,
    "mcp": false,
    "plugins": false,
    "skills": false
  }
}
```

## Suggested Fixes

1. **Honor OpenCode env flags**: Check `process.env.OPENCODE_DISABLE_CLAUDE_CODE` (and related vars) as hard gates in `src/plugin-config.ts` before evaluating `claude_code.*` config properties.
2. **Add global `claude_code.enabled` switch**: Single master toggle in `ClaudeCodeConfigSchema`, subordinating per-component switches.
3. **Safer default**: Auto-disable Claude plugin import when `OPENCODE_DISABLE_CLAUDE_CODE=1` is set.

## Related

- [anomalyco/opencode#10383](https://github.com/anomalyco/opencode/issues/10383) — Same issue reported against OpenCode core, closed as oh-my-opencode's responsibility.

### PR DESCRIPTION
**Pull request by gustavosmendes**
## Summary

Fixes #2037 — oh-my-opencode ignored `OPENCODE_DISABLE_CLAUDE_CODE` and related env vars, injecting Claude Code components into the resolved config regardless of the user's OpenCode-level opt-out.

### What changed

- **`applyEnvVarOverrides(config, env?)`** — pure function (env injected as parameter, testable without global mocks) called at the end of `loadPluginConfig`. Honors both master and per-component env vars:
  - `OPENCODE_DISABLE_CLAUDE_CODE=1|true` → disables all seven component gates
  - `OPENCODE_DISABLE_CLAUDE_CODE_SKILLS=1|true` → disables only skills (matching OpenCode core's per-component cascade from `flag.ts`)
- **`isTruthy(env, key)`** — case-insensitive env var parser (`"1"`, `"true"`, `"TRUE"`, `"True"` all work)
- **`isClaudeCodeEnabled(claudeCode, component)`** — consolidates the `?? true` / `!== false` scatter across 8 call sites into a single function. Also honours the new `claude_code.enabled` master switch.
- **`claude_code.enabled?: boolean`** field in `ClaudeCodeConfigSchema` — single JSON-level master toggle subordinating all per-component flags.

## Files changed

| File | Change |
|---|---|
| `src/config/schema/claude-code.ts` | `enabled?: boolean` field + `isClaudeCodeEnabled` helper |
| `src/plugin-config.ts` | `applyEnvVarOverrides` with `isTruthy` + call in `loadPluginConfig` |
| `src/plugin-handlers/plugin-components-loader.ts` | gate → `isClaudeCodeEnabled` |
| `src/plugin-handlers/command-config-handler.ts` | 2 gates → `isClaudeCodeEnabled` |
| `src/plugin-handlers/agent-config-handler.ts` | 2 gates → `isClaudeCodeEnabled` |
| `src/plugin-handlers/mcp-config-handler.ts` | gate → `isClaudeCodeEnabled` |
| `src/plugin/hooks/create-transform-hooks.ts` | gate → `isClaudeCodeEnabled` |
| `src/plugin/skill-context.ts` | `!== false` → `isClaudeCodeEnabled` |
| `src/plugin-config.test.ts` | 30 tests (84 assertions) covering overrides, case sensitivity, edge cases |

## Behavior

| Scenario | Before | After |
|---|---|---|
| `OPENCODE_DISABLE_CLAUDE_CODE=1` | All 7 gates active | All 7 gates disabled |
| `OPENCODE_DISABLE_CLAUDE_CODE=TRUE` | All 7 gates active | All 7 gates disabled (case-insensitive) |
| `OPENCODE_DISABLE_CLAUDE_CODE_SKILLS=1` | Skills active | Skills disabled, other 6 unchanged |
| `OPENCODE_DISABLE_CLAUDE_CODE=1` + `SKILLS=0` | N/A | All 7 disabled (master wins) |
| `claude_code.enabled: false` in JSON | N/A (field didn't exist) | All 7 gates disabled |
| `claude_code.mcp: false` in JSON | `mcp` disabled | unchanged |
| No env var, no JSON override | All 7 active | unchanged |

Existing workaround (`claude_code.{ agents,commands,hooks,mcp,plugins,skills }: false`) continues to work.

## Test plan

- [x] `bun test src/plugin-config.test.ts` — 30 pass (84 assertions)
- [x] `bun test` — full suite passes, 0 fail
- [x] Typecheck clean
- [ ] Manual: `OPENCODE_DISABLE_CLAUDE_CODE=1 opencode debug config | rg "everything-claude-code"` should return empty

## Review feedback addressed

- **cubic P1**: Added `OPENCODE_DISABLE_CLAUDE_CODE_SKILLS` per-component support (confirmed real env var from OpenCode core `flag.ts`)
- **Copilot**: Fixed docstring from "pure" to "testable without global mocks"
- **Copilot**: Added per-component env var support (`SKILLS`)
- **Code quality**: Removed double shallow-copy at call site
- **Testing**: Added case-insensitive tests, edge cases (`""`, `"yes"`, `undefined`), per-component override tests

### CODE REVIEW

File: src/plugin-config.ts [hunk 1]
```diff
@@ -11,8 +11,41 @@ import {
...
+ * Accepts an optional `env` parameter so the function stays pure and testable.
+ */
+export function applyEnvVarOverrides(
+  config: OhMyOpenCodeConfig,
+  env: NodeJS.ProcessEnv = process.env,
+): OhMyOpenCodeConfig {
+  const masterDisabled =
+    env.OPENCODE_DISABLE_CLAUDE_CODE === "1" ||
+    env.OPENCODE_DISABLE_CLAUDE_CODE === "true";
+
```
[the remark this answers is not in the document]
Comment (gustavosmendes) [reply src/plugin-config.ts:55]: Addressed in c0ba654:

1. **`OPENCODE_DISABLE_CLAUDE_CODE_SKILLS`** is now honored — disables only the `skills` component. Master switch takes precedence when both are set.
2. **`OPENCODE_DISABLE_CLAUDE_CODE_PROMPT`** was intentionally omitted — it controls `~/.claude/CLAUDE.md` loading in OpenCode core, but oh-my-opencode doesn't load CLAUDE.md as a prompt, so there's no component to map it to.
3. **`OPENCODE_DISABLE_EXTERNAL_SKILLS`** cascades from `SKILLS` in OpenCode core and is beyond this plugin's scope (external skill loading is handled by OpenCode itself).
4. Parsing is now case-insensitive via `isTruthy()` helper (`"1"`, `"true"`, `"TRUE"`, `"True"` all work).

File: src/plugin-config.ts [hunk 2]
```diff
@@ -11,8 +11,41 @@ import {
...
+} as const;
+
+/**
+ * Applies hard overrides from OpenCode env vars onto the resolved config.
+ * Accepts an optional `env` parameter so the function stays pure and testable.
+ */
+export function applyEnvVarOverrides(
+  config: OhMyOpenCodeConfig,
+  env: NodeJS.ProcessEnv = process.env,
+): OhMyOpenCodeConfig {
```
[the remark this answers is not in the document]
Comment (gustavosmendes) [reply src/plugin-config.ts:45]: Addressed in c0ba654 — docstring updated from "stays pure" to "testable without global mocks", which accurately describes the design intent (env parameter injection for testing, `process.env` default for production).

**Comment by code-yeongyu** [comment]
[sisyphus-bot]

## PR Review Assessment

### Summary
This PR fixes the `OPENCODE_DISABLE_CLAUDE_CODE` environment variable being ignored by oh-my-opencode. The fix properly honors both the master switch and per-component overrides.

### Changes Analysis
| Aspect | Status |
|--------|--------|
| **CI Status** | ✅ All passing (test, typecheck, build, cla, GitGuardian) |
| **Test Coverage** | ✅ 30 new tests (84 assertions) covering env overrides, case-insensitivity, edge cases |
| **Code Quality** | ✅ Clean consolidation of 8 scattered gate checks into `isClaudeCodeEnabled()` helper |
| **Files Changed** | 9 files - focused, surgical changes |

### Technical Review

**What it fixes:**
- `OPENCODE_DISABLE_CLAUDE_CODE=1|true` now disables all 7 Claude Code component gates
- `OPENCODE_DISABLE_CLAUDE_CODE_SKILLS=1|true` disables only skills (per OpenCode core convention)
- Added `claude_code.enabled` master gate in schema
- Case-insensitive parsing (`"1"`, `"true"`, `"TRUE"` all work)

**Architecture:**
- `applyEnvVarOverrides()` is testable (env injected as parameter)
- Pure function approach avoids global mocks in tests
- 8 call sites now use single `isClaudeCodeEnabled()` helper instead of scattered `?? true` checks

**Review Feedback Addressed:**
- ✅ cubic-dev-ai P1 concern: `OPENCODE_DISABLE_CLAUDE_CODE_SKILLS` is now supported (see `src/plugin-config.ts` lines 62-70)
- ✅ Copilot docstring feedback addressed
- ✅ Code quality: removed double shallow-copy

### Risk Assessment
| Risk | Level | Notes |
|------|-------|-------|
| Regression | LOW | All existing tests pass; new tests cover edge cases |
| Breaking Change | NONE | Adds new capability without changing existing behavior |
| Scope Creep | NONE | Changes are focused and surgical |

### Recommendation
This is a well-structured bugfix with:
- Comprehensive test coverage
- Clean architectural approach
- All CI checks passing
- Review feedback incorporated

**Action Required:** Needs maintainer approval to merge. The `reviewDecision` field is empty and `mergeable` is UNKNOWN, so automatic merging is blocked pending human review.
